### PR TITLE
research(new-directions): V6 recurrence headroom — H0 refuted, conditioning takes most of it

### PR DESCRIPTION
**Pull request by techforky**
Executes the preregistration in #1875. Development diagnostic, permanently nonpromoting. Micro suite only — no IPMNIST lane, no pinned artifact, no registered source, no promotion seed, no change to `alberta_framework/`.

## Verdict: my registered prediction is REFUTED

I registered H0 — "no existing arm exploits recurrence" — and it is wrong. All six ladder arms clear the pre-registered criterion, mean paired gap `> 0` with all three seeds positive.

| arm | M1 `input_permutation` | M4 `recurrence` | gap |
|---|---|---|---|
| `sgd_raw` | 0.2827 | 0.6696 | **+0.3869** |
| `adamw` | 0.2281 | 0.5788 | **+0.3507** |
| `upgd_raw` | 0.2797 | 0.3726 | +0.0929 |
| `sgd_norm` | 0.6808 | 0.7270 | +0.0461 |
| `gated_norm` | 0.6911 | 0.7383 | +0.0472 |
| `naive_bayes` | 0.6052 | 0.6526 | +0.0474 |

**Where the reasoning failed.** I argued "none of these arms indexes permutations, therefore none can exploit recurrence." That conflates *having a recurrence mechanism* with *benefiting from recurrence*. No arm indexes permutations — this result does not show otherwise — but every arm carries persistent weights, and when a mapping repeats those weights are already partly fit. The benefit is implicit reuse, not recognition.

So **"exploits recurrence" here means "scores better when permutations repeat", not "implements a recurrence mechanism"**, and that distinction bounds every downstream claim.

## The structure is the finding

| group | mean gap |
|---|---|
| unconditioned (`sgd_raw`, `adamw`) | **+0.3688** |
| conditioned (`sgd_norm`, `gated_norm`, `naive_bayes`) | **+0.0469** |
| ratio | **7.9x** |

Arms that are bad on fresh permutations gain enormously from repetition; arms that are good on fresh permutations gain almost nothing. A conditioned arm re-adapts fast enough that a repeat is barely cheaper than a fresh permutation; an unconditioned arm re-adapts slowly and a repeat saves most of that cost.

This mirrors the campaign's central result — conditioning is the load-bearing mechanism — measured on the memory axis rather than the input-shift axis.

## What this does and does not bound for direction D

**Measured.** With 20x reuse (5 permutations across 100 regimes), a champion-like conditioned arm gains **+0.047**. That is what implicit weight retention already captures, and it is the baseline an explicit recurrence-indexing mechanism must beat rather than a starting point of zero.

**Not bounded here.** This does *not* cap direction D. An explicit mechanism could jump straight to stored per-permutation state rather than partially re-adapting, and would then beat implicit retention. The distance from the best M4 arm to Bayes is `0.9833 - 0.7383 = 0.2450`, but most of that is ordinary online estimation cost, so it bounds the direction only in the loosest sense.

## Controls

Run first, reported regardless:

| family | regimes | distinct permutations |
|---|---|---|
| M1 | 100 | 100 — all fresh |
| M4 | 100 | **5** — exactly `recurrence_pool` |

Had the families been secretly identical, every gap would be zero and would have read as clean support for H0 — indistinguishable from the experiment's own failure mode. That is why it ran first.

**Bayes accuracy is family-invariant at 0.9833** (`bayes_reference`, 200k samples, `mc_sem` 0.0003), identical for both families. I verified this rather than assuming it: a family-dependent ceiling would have made every number above meaningless.

## Deviation

Wall clock **1911.6 s (32 min)** against the ~14 min I estimated in the preregistration. The estimate timed `gated_norm`; the raw arms and `naive_bayes` are slower. No protocol term changed — only my cost estimate was wrong.

## Verification

Commit `07034c49baad9ca9de07a602366ae0a4b5a9d948`, based on `cc877f78`. 36 runs plus the control, using the shipped `micro_continual` runner with no new benchmark code. `ruff` clean.

## Note on ledger numbering

The new entry is numbered **16** against current `main`. #1873 (V5) also adds a 16; whichever lands first, the other needs a renumber. Happy to rebase.

---

AI provider/model: anthropic / claude-opus-5 (as reported by the running client)
Client / agent tooling: claude-code
Attribution status: self-reported.

### CODE REVIEW

**Comment by techforky** [comment]
## Scoping note: direction D does not apply to the IPMNIST lane

Recorded because it bounds what this PR's number is good for, and it partly undercuts the practical value of the work above.

`CEILING_ANALYSIS.md` lists recurrence-indexed context reuse (direction D) among the live candidates for attacking the champion's transient. But the IPMNIST protocol draws a fresh permutation every task:

```python
# upgd_ipmnist.build_schedule
permutations = jax.vmap(
    lambda task: jr.permutation(jr.fold_in(key_perm, task), config.input_dim)
)(tasks)
```

Measured at the published protocol (200 tasks, 784 dims):

```text
seed 0: 200 tasks -> 200 distinct permutations  repeats=0
seed 1: 200 tasks -> 200 distinct permutations  repeats=0
seed 2: 200 tasks -> 200 distinct permutations  repeats=0
```

Zero repeats, and structurally there can be none — the draw is uniform over 784! and independent per task.

**So a recurrence-indexed mechanism has nothing to index on this lane.** No amount of stored per-permutation state helps when no permutation ever returns.

### What that means for this PR

The +0.047 measured here is on the micro suite's M4 family, which is the only place in the repository where recurrence exists at all. It is a real number about recurrence; it is **not** a number about the champion lane's transient.

Combining with the result above, the scoping answer for direction D is:

1. **The IPMNIST lane has no recurrence**, so direction D cannot help the campaign's current transient at all.
2. **On the one lane that does have recurrence**, conditioned arms already capture most of the benefit implicitly — 20x reuse is worth +0.047 to `gated_norm`.
3. Therefore direction D would require both a *new recurring lane* and an explicit mechanism that beats the +0.047 implicit baseline, before it could pay anything.

That is a scoping conclusion about priority, **not** a claim that a recurrence mechanism is impossible or that direction D is refuted. It says the direction is inapplicable where the transient actually lives, and that where it is applicable, most of the easy benefit is already taken.

I would rather state this against my own work than leave the +0.047 to be read as headroom on the champion lane, which it is not.

**Review by lalalune — CHANGES_REQUESTED** [review]
Blocking evidence-contract issues:\n\n1. The preregistration says family separation is checked across the run, but the runner checks only . Add a post-hoc control audit for every exact seed and disclose the original execution deviation.\n2. The report claims a verified Bayes ceiling and computes headroom from 0.9833, but the runner/JSON contains no Bayes control. The repository’s three-seed mean differs from seed 0. Add per-seed Bayes raw values/MC SEM in a companion audit, aggregate on matched seeds, and recompute.\n3. The highlighted 7.9× grouping is post hoc, omits , and classifies  as conditioned despite its distinct mechanism. A smaller gap at a higher baseline is not a causal conditioning ablation. Keep the registered all-arm paired table primary and label any complete, predeclared taxonomy exploratory/descriptive.\n4. With three consumed development seeds, say the registered criterion was met in all three seeds; do not make an unqualified generalization.\n5. Bind exact arms, source/runner, dependencies/runtime, dataset/config/invocation, nonpromotion, and all raw/derived values in a versioned append-only validator.\n\nRebase after #1873 (renumber the conflicting ledger entry), retain the IPMNIST-nonrecurrence scope in-repository, and add tests that all controls complete before any run cell and any failure yields void/no accuracies.

**Review by lalalune — CHANGES_REQUESTED** [review]
Blocking evidence-contract issues:

1. The preregistration says family separation is checked across the run, but the runner checks only `args.seeds[0]`. Add a post-hoc control audit for every exact seed and disclose the original execution deviation.
2. The report claims a verified Bayes ceiling and computes headroom from 0.9833, but the runner/JSON contains no Bayes control. The repository’s three-seed mean differs from seed 0. Add per-seed Bayes raw values/MC SEM in a companion audit, aggregate on matched seeds, and recompute.
3. The highlighted 7.9× grouping is post hoc, omits `upgd_raw`, and classifies `naive_bayes` as conditioned despite its distinct mechanism. A smaller gap at a higher baseline is not a causal conditioning ablation. Keep the registered all-arm paired table primary and label any complete, predeclared taxonomy exploratory/descriptive.
4. With three consumed development seeds, say the registered criterion was met in all three seeds; do not make an unqualified generalization.
5. Bind exact arms, source/runner, dependencies/runtime, dataset/config/invocation, nonpromotion, and all raw/derived values in a versioned append-only validator.

Rebase after #1873 (renumber the conflicting ledger entry), retain the IPMNIST-nonrecurrence scope in-repository, and add tests that all controls complete before any run cell and any failure yields void/no accuracies.

**Comment by lalalune** [comment]
Superseded by the append-only, strict-audit replacement #1926. The original raw V6 files are preserved byte-for-byte; the replacement retains the observations only as an amended, inconclusive development result with all-seed controls and matched Bayes recomputation.
